clevertap-core/src/main/java/com/clevertap/android/sdk/utils/JsonUtils.kt

```diff
@@ -1,9 +1,39 @@
 package com.clevertap.android.sdk.utils
 
+import android.os.Parcel
+import org.json.JSONException
 import org.json.JSONObject
 
 fun JSONObject.putObject(name: String, init: JSONObject.() -> Unit) {
     val obj = JSONObject()
     init(obj)
     put(name, obj)
 }
+
+fun JSONObject.getStringOrNull(name: String): String? {
+    return if (has(name)) {
+        getString(name)
+    } else {
+        null
+    }
+}
+
+fun JSONObject.getObjectOrNull(name: String): JSONObject? {
+    return try {
+        getJSONObject(name)
+    } catch (je: JSONException) {
+        null
+    }
+}
```

Comment:
these methods might be present in JSONObject itself.

Reply:
I will remove getObjectOrNull since there is JSONObject.optJSONObject which behaves the same.  JSONObject.optString() however returns an empty string if the key was not found instead of null and I think getStringOrNull is useful.